consensus/blake3pow/consensus.go

```diff
@@ -384,11 +376,16 @@ func (blake3pow *Blake3pow) verifySeal(chain consensus.ChainHeaderReader, header
 	if header.Difficulty().Sign() <= 0 {
 		return errInvalidDifficulty
 	}
-	// Check that SealHash meets the difficulty target
-	target := new(big.Int).Div(big2e256, header.Difficulty())
-	if new(big.Int).SetBytes(header.Hash().Bytes()).Cmp(target) > 0 {
+	// Check for valid zone share and order matches context
+	order := header.CalcOrder()
```

Comment:
I would recommend CalcOrder just return the error instead of using -1 as a sentinel
https://github.com/dominant-strategies/go-quai/issues/748

Reply:
I can do this in a separate commit if we want, and just merge this. Because there are some other places that use CalcOrder as it is.

Reply:
calc order returns an error now